opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -118,9 +135,10 @@ class BoundedAttributes(MutableMapping):
 
     def __init__(
         self,
-        maxlen: Optional[int] = _DEFAULT_LIMIT,
+        maxlen: Optional[int] = None,
```

Comment:
A default value of 128 was masking multiple issues with this and even fooling tests. I don't think this should have a default as different types of attributes (links, events, spans, metrics, logs) could have different defaults. This is a behaviorally breaking change though so probably we should leave it as is. I'm a bit torn on this but leaning towards removing the default.

Reply:
Defaults and user provided values (via code or env) are handled by SpanLimits and passed down to BoundedAttribute as the first argument. This was duplicating default value at a completely different place which ended up masking some issues. For example, links were not passing the actual limit value to this class but it was working when users didn't configure anything as the argument defaulted to 128. 

I don't think this class should have any default values for limits and should expect them to be passed down by whatever is using it.